cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -90,10 +90,12 @@ func (c *CStorPoolController) cStorPoolEventHandler(operation common.QueueOperat
 		b) Watcher could have come up first, in this case, there is a possibility
 		that zrepl goes down and comes up and the watcher sees that no pool is there,
 		so it will break the loop and attempt to import the pool. */
-		existingPool, _ := pool.GetPoolName()
-		flag := len(existingPool) != 0
+
 		// cnt is no of attempts to wait and handle in case of already present pool.
 		cnt := common.NoOfPoolWaitAttempts
+		existingPool, _ := pool.GetPoolName()
+		flag := len(existingPool) != 0
+		common.Mux.Lock()
 		for i := 0; flag && i < cnt; i++ {
 			common.InitialImportedPoolVol, _ = volumereplica.GetVolumes()
```

Comment:
Add comment to specify the reason for initialising this variable with existing volume list.

Reply:
will address